clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchIndex.scala

```diff
@@ -75,28 +85,31 @@ object ElasticsearchIndex extends ModelAutoDerivation {
     DateTimeFormatter.ofPattern("yyyy/MM/dd")
 
   /** Implicit summoner, for convenience. */
-  def apply[Document: ElasticsearchIndex]: ElasticsearchIndex[Document] =
-    implicitly[ElasticsearchIndex[Document]]
+  def apply[Index]: ElasticsearchIndex[Index] =
+    implicitly[ElasticsearchIndex[Index]]
 
-  implicit val WgsUbam: ElasticsearchIndex[DocumentWgsUbam] =
-    new ElasticsearchIndex[DocumentWgsUbam](
+  implicit val WgsUbam: ElasticsearchIndex[WgsUbamIndex.type] =
+    new ElasticsearchIndex(
       "wgs-ubam",
+      WgsUbamIndex,
       ElasticsearchFieldMapper.StringsToTextFieldsWithSubKeywords
     )
 
-  implicit val Gvcf: ElasticsearchIndex[DocumentGvcf] =
+  implicit val Gvcf: ElasticsearchIndex[GvcfIndex.type] =
     // Despite being decoupled from "v1", we append -v2 to keep ES indices consistent with GCS.
     // Since we compute GCS paths from the ES index name, inconsistency would break GCS paths.
-    new ElasticsearchIndex[DocumentGvcf](
+    new ElasticsearchIndex(
       "gvcf-v2",
+      GvcfIndex,
       ElasticsearchFieldMapper.StringsToTextFieldsWithSubKeywords
     )
 
-  implicit val WgsCram: ElasticsearchIndex[DocumentWgsCram] =
+  implicit val WgsCram: ElasticsearchIndex[WgsCramIndex.type] =
```

Comment:
Since `WgsCramIndex` is an argument to the constructor, is there a way to avoid having to pass `.type` as a type parameter too?

Reply:
Nope, compiler complains that `ElasticsearchIndex` needs type parameters.

Reply:
Did you try removing the type parameter from `ElasticsearchIndex` and using `index.type` in the code? It seems like there should be a way to avoid duplicating the arguments here. In other cases we were able to migrate type parameters to fields on `Index` objects.

Reply:
Removing the type parameter from `ElasticsearchIndex` entirely?

Reply:
We could probably make that work... though it doesn't pair as nicely with Charley's in-flight changes. The idea was that we'd have a (simplified) signature like this for the service classes:
```scala
class IndexService[TI <: TransferIndex](index: TI, esIndex: ElasticsearchIndex[TI])
```
So you can bind the two together a bit.

Reply:
If it's not ok to do it now, we can save this for a future refactor.